src/cashlib/cashlib.cpp

```diff
@@ -0,0 +1,218 @@
+/* clang-format on */
+#include "base58.h"
+#include "random.h"
+#include "streams.h"
+#include "utilstrencodings.h"
+
+#include <openssl/rand.h>
+#include <string>
+#include <vector>
+
+static bool sigInited = false;
+
+// stop the logging
+int LogPrintStr(const std::string &str) { return str.size(); }
+namespace Logging
+{
+uint64_t categoriesEnabled = 0; // 64 bit log id mask.
+};
+
+// helper functions
+namespace
+{
+#if 0
+CKey LoadKey(uint256 val)
+{
+    CKey secret;
+    unsigned char *src = val.begin();
+    secret.Set(src, src+32, true);
+    return secret;
+}
+#endif
+
+CKey LoadKey(unsigned char *src)
+{
+    CKey secret;
+    secret.Set(src, src + 32, true);
+    return secret;
+}
+
+// From core_read.cpp #include "core_io.h"
+bool DecodeHexTx(CTransaction &tx, const std::string &strHexTx)
+{
+    if (!IsHex(strHexTx))
+        return false;
+
+    std::vector<unsigned char> txData(ParseHex(strHexTx));
+    CDataStream ssData(txData, SER_NETWORK, PROTOCOL_VERSION);
+    try
+    {
+        ssData >> tx;
+    }
+    catch (const std::exception &)
+    {
+        return false;
+    }
+
+    return true;
+}
+
+#if 0
+bool DecodeHexTx(CTransaction &tx, const char* cHexTx)
+{
+    std::string strHexTx(cHexTx);
+    return DecodeHexTx(tx, strHexTx);
+}
+
+std::string GetHexReversed(const std::vector<unsigned char>& data)
+{
+    unsigned int len = data.size();
+    char psz[len * 2 + 1];
+    for (unsigned int i = 0; i < len; i++)
+        sprintf(psz + i * 2, "%02x", data[len - i - 1]);
+    return std::string(psz, psz + len * 2);
+}
+
+std::string GetHex(const std::vector<unsigned char>& data)
+{
+    unsigned int len = data.size();
+    char psz[len * 2 + 1];
+    for (unsigned int i = 0; i < len; i++)
+        sprintf(psz + i * 2, "%02x", data[i]);
+    return std::string(psz, psz + len * 2);
+}
+#endif
+
+static const char *hexxlat = "0123456789ABCDEF";
+std::string GetHex(unsigned char *data, unsigned int len)
+{
+    std::string ret;
+    ret.reserve(2 * len);
+    for (unsigned int i = 0; i < len; i++)
+    {
+        unsigned char val = data[i];
+        ret.push_back(hexxlat[val >> 4]);
+        ret.push_back(hexxlat[val & 15]);
+    }
+    return ret;
+}
+}
+
+#if 0
+extern "C" int Base58Key2Bin(char* b58key, unsigned char* result)
+{
+    CBitcoinSecret s;
+    s.SetString(b58key);
+    CKey key = s.GetKey();
+
+    unsigned char *src = (unsigned char *)key.begin();
+    memcpy(result, src, 32);
+    return 32;
+}
+#endif
+
+/** Convert binary data to a hex string.  The provided result buffer must be 2*length+1 bytes.
+ */
+extern "C" int Bin2Hex(unsigned char *val, int length, char *result, unsigned int resultLen)
+{
+    std::string s = GetHex(val, length);
+    if (s.size() >= resultLen)
+        return 0; // need 1 more for /0
+    strncpy(result, s.c_str(), resultLen);
+    return 1;
+}
+
+
+/** Return random bytes from cryptographically acceptable random sources */
+extern "C" int RandomBytes(unsigned char *buf, int num)
```

Comment:
why not using `GetRandBytes` defined in `random.cpp`

Reply:
It logs and asserts.  Also it is name mangled, so would have to wrap anyway.  This routine returns the number of random bytes generated, so that result can be checked and a higher level language exception can be thrown.